camel/agents/embodied_agent.py

```diff
@@ -169,20 +100,30 @@ def step(
 
         # NOTE: Only single output messages are supported
         output_message = output_messages[0]
-        explanation, code = self.get_explanation_and_code(
-            output_message.content)
-        if self.verbose:
-            print_text_animated(self.logger_color +
-                                f"> Explanation:\n{explanation}")
-            print_text_animated(self.logger_color + f"> Code:\n{code}")
+        explanations_list, codes_list =\
+            output_message.extract_text_and_code_prompts()
 
-        if code is not None:
-            global_vars = {action.name: action for action in self.action_space}
-            executed_results = self.execute_code(code, global_vars)
-            output_message.content += (
-                f"\n> Executed Results:\n{executed_results}")
+        if self.verbose:
+            for explanation, code in zip(explanations_list, codes_list):
+                print_text_animated(self.logger_color +
+                                    f"> Explanation:\n{explanation}")
+                print_text_animated(self.logger_color + f"> Code:\n{code}")
+
+            if len(explanation) > len(code):
+                print_text_animated(self.logger_color +
+                                    f"> Explanation:\n{explanation}")
+
+        if codes_list is not None:
+            output_message.content = "\n> Executed Results:"
+            for code in codes_list:
+                global_vars = {
+                    action.name: action
+                    for action in self.action_space
+                }
+                executed_results = code.execute(global_vars)
+                output_message.content += (f"\n{executed_results}")
```

Comment:
We can improving the output format a bit:
```suggestion
                executed_outputs = code.execute(global_vars)
                output_message.content += (f"- Python standard output:\n{executed_outputs[0])}\n"
                    f"- Local variables:\n{executed_outputs[1]}\n")
                output_message.content += "*" * 50 + "\n"
```

Reply:
Thank you for the improvements. Let me merge it.